src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -386,6 +386,7 @@ public void Stake(WalletSecret walletSecret)
             if (Interlocked.CompareExchange(ref this.stakeProgressFlag, StakeInProgress, StakeNotInProgress) == StakeInProgress)
```

Comment:
Since this block of code is being changed, perhaps this if condition could be given more meaning?   My guess (from not knowing what this code does) but merely from the log message that it means   if (this.StakingInProgress())

Reply:
This is a good comment, but it should be part of https://github.com/stratisproject/StratisBitcoinFullNode/issues/1397

It does not really belong here. Needless to say that `StakingInProgress` would not be a good name. So we need to think about a good name, but 1st we need to fix the race condition, which may actually cause this code to be removed or 
rewritten significantly.

Reply:
>but 1st we need to fix the race condition, which may actually cause this code to be removed or
rewritten significantly.

Sounds good - but can I actually put this comment on #1397 ?

Reply:
Absolutely, it is most relevant there